### PR TITLE
Move the remaining flag functions into flagutil

### PR DESCRIPTION
**Pull request by tempoz**
Also guard against the `config_file` flag being present in the YAML
config, as it is unsupported.

---

**Version bump**: Patch

### CODE REVIEW

File: server/config/config.go [hunk 1]
```diff
@@ -2,53 +2,18 @@ package config
...
-		return err
-	}
-	if err := flagutil.RetypeAndFilterYAMLMap(yamlMap, typeMap, []string{}); err != nil {
-		return fmt.Errorf("Error encountered retyping YAML map: %s", err)
-	}
+var configPath = flag.String(pathFlagName, "/config.yaml", "The path to a buildbuddy config file")
 
-	return flagutil.PopulateFlagsFromYAMLMap(yamlMap)
+func init() {
+	flagutil.IgnoreFlag(pathFlagName)
```
Comment (tylerwilliams) [inline server/config/config.go:14]: nit: can you add a comment explaining why we do this?

**Review by tylerwilliams — APPROVED** [review]
LG (1 nit)

**Comment by tempoz** [comment]
Current dependencies on/for this PR:
* master
  * **PR #1937** <a href="https://app.graphite.dev/github/pr/buildbuddy-io/buildbuddy/1937" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/buildbuddy-io/buildbuddy/1947?utm_source=stack-comment).

File: server/config/config.go (hunk 1 above)
Comment (tempoz) [reply server/config/config.go:14]: Done!

**Pull request merged**